### ISSUE TITLE
bracket matching is slower than Atom's builtin version

### ISSUE THREAD

**Issue by trusktr** [t9md/atom-vim-mode-plus#1069]
# Check list

You have to check all before open issue.

- [x] Provide your environment info clipped by `Vim Mode Plus: Clip Debug Info` command.

<details>
  <summary>debug info</summary>

```json
{
  "atom": "1.27.0",
  "platform": "linux",
  "release": "4.13.0-41-generic",
  "vmpVersion": "1.32.0",
  "vmpConfig": {
    "blackholeRegisteredOperators": [],
    "startInInsertModeScopes": [],
    "charactersToAddSpaceOnSurround": [],
    "highlightSearchExcludeScopes": [],
    "flashOnOperateBlacklist": []
  }
}
```

</details>

- [x] Try with latest Atom and latest vim-mode-plus.
- [x] Pick descriptive and non-ambiguous subject
- [x] Express "what" you want(feature? config option?, behavior change?) in short sentence(not long!).

Want `%` bracket matching to be fast. It is slow compared to Atom's `ctrl-m` in big files. It takes a whole second, but Atom's is instant.

Maybe `%` should be just a small tweak to Atom's builtin, if possible?

- [ ] Contrast current behavior if you want to change current behavior, with sample text, operation(keystroke) and result.
- [ ] Include real use case so that maintainer can understand "why" you need help.

**Comment by trusktr** [comment]
@t9md Have you noticed this? In large files, the `%` feature is unbearably slow, while Atom's `ctrl+m` feature is perceivably instantaneous.

**Comment by keevan** [comment]
> @t9md Have you noticed this? In large files, the `%` feature is unbearably slow, while Atom's `ctrl+m` feature is perceivably instantaneous.

@trusktr Depending on what you are coding, you might not need to use VMP's bracket matching feature. I however use it because I find myself jumping between brackets *within a string*, which Atom does not seem to support out of the box using `ctrl+m`

**Comment by t9md** [comment]
@trusktr Yes, I've noticed. You are right. And I know why vmp is much slower than brackets-matcher.
Maybe I can improve speed, but I assure vmp would still slower than bracket-matcher's one.
I agree vmp's `%` is unacceptably slow in some codebase.
I will keep open this issue, but nothing planned to tackle this issue for now.
